docs/LogGuide.md

```diff
@@ -1706,14 +1747,36 @@ at the moment the action is "locked in".
 > so I would need to do a lot of work-arounds."
 
 Structure:
-`25:[Player ObjectId]:[Sequence Number]:[Current HP]:[Max HP]:[Current MP]:[Max MP]:[Current TP]:[Max TP]:[Position X]:[Position Y]:[Position Z]:[Facing]:[packet data thereafter]`
-
+`25:[Player ObjectId]:[Sequence Number]:[Current HP]:[Max HP]:[Current MP]:[Max MP]:[Unused]:[Unused]:[Position X]:[Position Y]:[Position Z]:[Facing]:[packet data thereafter]`
+`37|[timestamp]|[targetId]|[targetName]|[Sequence Number]|[target hp]|[target max hp]|[target mp]|[target max mp]|[unused]|[unused]|[x]|[y]|[z]|[heading]|[unknown]|[unknown]|[unknown]|[unknown]|[unknown]|[unknown]|[unknown]|[unknown]|16985ce13b1e6fa6`
 Examples:
 
 ```log
 25:12345678:PlayerOne:0000132A:33635:35817:10000:10000:0::0.3841706:-207.8767:2.901163:-3.00212:03E8:2500:0:01:03000000:0:0:E0000000:
+37|2022-02-12T20:06:22.0750000-08:00|1234ABCD|Foo Bar|0004ADA6|473|473|73|10000|0||38.99|101.76|-8.00|1.11|0800|0|04DE|01|020008B2|BFD20004|0|107284AA|16985ce13b1e6fa6
 ```
 
+#### Tracking Ability Resolution
+
+Unfortunately, this is not particularly clean. For one, while the server does tell the game when an action is resolved,
+it does not tell the game when an action will not resolve (ghosting). However, the caster dying or target becoming
+untargetable is usually a decent indicator that something will not resolve.
+
+Note that AoE abilities may have the same sequence ID for all targets hit. Thus, you need to key off of both the
+sequence ID, *and* the target.
+
+In addition, you don't always get a 37-line. For example, a heal that is pure overheal will not generate one.
```

Comment:
Just track action events without a sequence confirmation? You can do this reliably without any dodgy hacks.

Reply:
But how do you differentiate between ghosted and pending if you don't track them?

Reply:
When you have both events, the sequence IDs correlate. Sequence IDs without a correlated result are ghosted. You can watch for events with a 3 second buffer (maximum event interval) and if there's no result you know it was ghosted.

Reply:
Oh, I see what you mean, but that still means you're incorrect for that window if something dies before actions resolve. Probably not a huge deal, but the goal behind updating this doc is to better answer the question of "how do I take a log and convert it back to a reasonably abstracted game state".

Reply:
If the target dies, you still get the sequence followup, the actor ID is just set to 0xE0000000 which is the game internal way of representing an actor that's no longer instantiated, unless I forgot how this works but that should be correct.

Reply:
I don't think that's accurate. Easy enough to test on WHM. Find a low level mob that would get oneshot by assize. Cast Holy  then immediately assize so that the Holy doesn't have a chance to resolve properly.
```
# Holy snapshot
21|2022-04-08T23:29:52.5510000-07:00|108AA260|Wynn Dohz|6504|Holy III|40029A6C|Hornet Swarm|750003|F5B0000|E|20000|1B|65048000|0|0|0|0|0|0|0|0|0|0|217|217|0|10000|||181.32|364.43|-32.38|0.00|50437|50437|9600|10000|||181.79|363.91|-32.36|-2.82|00095A40|0|1|8228a170257c3adf
# Assize heal snapshot
22|2022-04-08T23:29:53.1760000-07:00|108AA260|Wynn Dohz|DF3|Assize|108AA260|Wynn Dohz|10004|32680000|B|1F40000|0|0|0|0|0|0|0|0|0|0|0|0|50437|50437|9600|10000|||181.79|363.91|-32.36|-2.82|50437|50437|9600|10000|||181.79|363.91|-32.36|-2.82|00095A42|0|2|0d6958009f133a27
# Assize damage snapshot
22|2022-04-08T23:29:53.1760000-07:00|108AA260|Wynn Dohz|DF3|Assize|40029A6C|Hornet Swarm|750003|2B750000|0|0|0|0|0|0|0|0|0|0|0|0|0|0|217|217|0|10000|||181.32|364.43|-32.38|0.00|50437|50437|9600|10000|||181.79|363.91|-32.36|-2.82|00095A42|1|2|ea150610e59dce66
# Heal resolves
37|2022-04-08T23:29:53.8400000-07:00|108AA260|Wynn Dohz|00095A42|50437|50437|10000|10000|0||181.79|363.91|-32.36|-2.82|1800|0|BF3D|0|c249768f640e5a2b
# Damage resolves
37|2022-04-08T23:29:53.9730000-07:00|40029A6C|Hornet Swarm|00095A42|0||||||181.32|364.43|-32.38|0.00|b50a4b060fc13a33
```
In addition, even if it did generate a 37-line with E0... as the target, you still wouldn't know which ones snapshotted or ghosted on a per-target basis.